commands/payment_channel_daemon_test.go

```diff
@@ -1,547 +1,598 @@
-	require.NoError(err)
-	target, err := address.NewFromString(fixtures.TestAddresses[1])
-	require.NoError(err)
-
-	eol := types.NewBlockHeight(20)
-	amt := types.NewAttoFILFromFIL(10000)
-
-	targetDaemon := th.NewDaemon(
-		t,
-		// must include 0th keyfilepath if using 0th TestMiner
-		th.WithMiner(fixtures.TestMiners[0]),
-		th.KeyFile(fixtures.KeyFilePaths()[0]),
-		th.KeyFile(fixtures.KeyFilePaths()[1]),
-	).Start()
-	defer targetDaemon.ShutdownSuccess()
-
-	daemonTestWithPaymentChannel(t, &payer, &target, amt, eol, func(d *th.TestDaemon, channelID *types.ChannelID) {
-		assert := assert.New(t)
-
-		d.ConnectSuccess(targetDaemon)
-
-		voucher := createVoucherStr(t, d, channelID, types.NewAttoFILFromFIL(111), &payer, uint64(0))
+	// This test should run in 20 block times
+	ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(20*series.GlobalSleepDelay))
+	defer cancel()
 
-		mustRedeemVoucher(t, targetDaemon, voucher, &target)
+	// Get basic testing environment
+	env, _, NewNode, ctx := fasting.BasicFastSetup(ctx, t, fast.EnvironmentOpts{})
 
-		ls := listChannelsAsStrs(targetDaemon, &payer)[0]
-		assert.Equal(fmt.Sprintf("%v: target: %s, amt: 10000, amt redeemed: 111, eol: 20", channelID.String(), target.String()), ls)
-	})
-}
+	// Teardown after test ends
+	defer func() {
+		err := env.Teardown(ctx)
+		require.NoError(err)
+	}()
 
-func TestPaymentChannelRedeemTooEarlyFails(t *testing.T) {
-	t.Parallel()
-	require := require.New(t)
+	// Start test
+	targetDaemon := NewNode()
+	payerDaemon := NewNode()
 
-	payer, err := address.NewFromString(fixtures.TestAddresses[2])
-	require.NoError(err)
-	target, err := address.NewFromString(fixtures.TestAddresses[1])
+	addrs, err := targetDaemon.AddressLs(ctx)
 	require.NoError(err)
+	targetAddr := addrs[0]
 
-	eol := types.NewBlockHeight(20)
-	amt := types.NewAttoFILFromFIL(10000)
+	addrs, err = payerDaemon.AddressLs(ctx)
+	require.NoError(err)
+	payerAddr := addrs[0]
 
-	targetDaemon := th.NewDaemon(
-		t,
-		// must include 0th keyfilepath if using 0th TestMiner
-		th.WithMiner(fixtures.TestMiners[0]),
-		th.KeyFile(fixtures.KeyFilePaths()[0]),
-		th.KeyFile(fixtures.KeyFilePaths()[1]),
-	).Start()
-	defer targetDaemon.ShutdownSuccess()
+	channelExpiry := types.NewBlockHeight(20)
+	channelAmount := types.NewAttoFILFromFIL(1000)
 
-	daemonTestWithPaymentChannel(t, &payer, &target, amt, eol, func(d *th.TestDaemon, channelID *types.ChannelID) {
-		assert := assert.New(t)
+	WithPaymentChannel(ctx, t, payerDaemon, payerAddr, targetAddr, channelAmount, channelExpiry, func(chanid *types.ChannelID) {
+		voucherAmount := types.NewAttoFILFromFIL(10)
+		voucherValidAt := types.NewBlockHeight(0)
+		voucherStr, err := payerDaemon.PaychVoucher(ctx, chanid, voucherAmount, fast.AOFromAddr(payerAddr), fast.AOValidAt(voucherValidAt))
+		require.NoError(err)
 
-		d.ConnectSuccess(targetDaemon)
+		// TODO(tperson) allow setting Price / Limit on process?
```

Comment:
What does this mean? A change to FAST or the test?

Reply:
FAST it's a place holder comment for myself to address https://github.com/filecoin-project/go-filecoin/issues/1811